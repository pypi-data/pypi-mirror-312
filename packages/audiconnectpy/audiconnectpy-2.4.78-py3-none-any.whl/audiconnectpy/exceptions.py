"""Exceptions for Audi connect."""


class AudiException(Exception):
    """General exception."""


class AuthorizationError(AudiException):
    """Authentication error."""


class HttpRequestError(AudiException):
    """HTTP Requests error."""


class TimeoutExceededError(AudiException):
    """Timeout exceeded."""


class ServiceNotFoundError(AudiException):
    """Service not found."""
