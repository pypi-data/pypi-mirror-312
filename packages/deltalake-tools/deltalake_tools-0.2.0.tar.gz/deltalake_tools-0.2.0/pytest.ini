[pytest]
log_cli = True
filterwarnings =
    ignore:.*
