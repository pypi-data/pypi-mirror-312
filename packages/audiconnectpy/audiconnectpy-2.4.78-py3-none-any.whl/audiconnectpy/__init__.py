"""audiconnectpy package."""

from .api import MODELS, AudiConnect
from .exceptions import AudiException, AuthorizationError

__all__ = ["AudiConnect", "AudiException", "AuthorizationError", "MODELS"]
