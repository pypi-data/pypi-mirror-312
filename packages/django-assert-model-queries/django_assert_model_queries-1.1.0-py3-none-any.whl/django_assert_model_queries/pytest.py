import pytest

from .test import AssertModelQueries


@pytest.fixture(scope="session")
def assert_model_queries():
    """
    An assertion fixture that is used as a context manager.

    Usage:

        def test_something(self, assert_model_queries):
            with assert_model_queries({"MyModel": 1}):
                do_something()

    """
    return AssertModelQueries()
