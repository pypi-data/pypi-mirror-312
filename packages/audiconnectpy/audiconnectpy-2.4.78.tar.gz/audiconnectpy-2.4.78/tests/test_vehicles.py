"""Tests analytics."""

from __future__ import annotations

import logging
from unittest.mock import patch

from aiohttp import ClientSession
import pytest
from syrupy.assertion import SnapshotAssertion

from audiconnectpy import AudiConnect, AudiException

USR = "x.y@z.zz"
PWD = "password"
COUNTRY = "FR"
SPIN = 1234

_LOGGER = logging.getLogger(__name__)


@patch("audiconnectpy.auth.Auth.async_connect")
@patch("audiconnectpy.api.AudiConnect._async_fill_url")
async def test_vehicle_0(
    connect,
    fill_url,
    snapshot: SnapshotAssertion,
    information_vehicles,
    vehicle_0,
    position,
    location,
    capabilities,
) -> None:
    """Test connection."""
    api = AudiConnect(
        session=ClientSession(), username=USR, password=PWD, country=COUNTRY, spin=SPIN
    )

    with (
        patch(
            "audiconnectpy.api.AudiConnect.async_get_information_vehicles",
            return_value=information_vehicles,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_selectivestatus",
            return_value=vehicle_0,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_position",
            return_value=position,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_location",
            return_value=location,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_capabilities",
            return_value=capabilities,
        ),
    ):
        await api.async_login()
        my_vehicle = api.vehicles[0]
        assert api.vehicles is not None
        assert my_vehicle.infos.to_dict() == snapshot
        assert my_vehicle.access.to_dict() == snapshot
        assert my_vehicle.fuel_status.to_dict() == snapshot
        assert my_vehicle.vehicle_health_inspection.to_dict() == snapshot
        assert my_vehicle.vehicle_lights.to_dict() == snapshot
        assert my_vehicle.measurements.to_dict() == snapshot
        assert my_vehicle.oil_level.to_dict() == snapshot
        assert my_vehicle.vehicle_health_warnings.to_dict() == snapshot


@patch("audiconnectpy.auth.Auth.async_connect")
@patch("audiconnectpy.api.AudiConnect._async_fill_url")
async def test_vehicle_1(
    connect,
    fill_url,
    snapshot: SnapshotAssertion,
    information_vehicles,
    vehicle_1,
    position,
    location,
    capabilities,
) -> None:
    """Test connection."""
    api = AudiConnect(
        session=ClientSession(), username=USR, password=PWD, country=COUNTRY, spin=SPIN
    )

    with (
        patch(
            "audiconnectpy.api.AudiConnect.async_get_information_vehicles",
            return_value=information_vehicles,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_selectivestatus",
            return_value=vehicle_1,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_position",
            return_value=position,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_location",
            return_value=location,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_capabilities",
            return_value=capabilities,
        ),
    ):
        await api.async_login()
        my_vehicle = api.vehicles[0]

        assert api.vehicles is not None
        assert my_vehicle.infos.to_dict() == snapshot
        assert my_vehicle.user_capabilities.to_dict() == snapshot
        assert my_vehicle.access.to_dict() == snapshot
        assert my_vehicle.charging.to_dict() == snapshot
        assert my_vehicle.climatisation_timers.to_dict() == snapshot
        assert my_vehicle.climatisation.to_dict() == snapshot
        assert my_vehicle.fuel_status.to_dict() == snapshot
        assert my_vehicle.vehicle_health_inspection.to_dict() == snapshot
        assert my_vehicle.vehicle_lights.to_dict() == snapshot
        assert my_vehicle.measurements.to_dict() == snapshot
        # assert my_vehicle.oil_level.to_dict() == snapshot
        # assert my_vehicle.vehicle_health_warnings.to_dict() == snapshot


@patch("audiconnectpy.auth.Auth.async_connect")
@patch("audiconnectpy.api.AudiConnect._async_fill_url")
async def test_vehicle_2(
    connect,
    fill_url,
    snapshot: SnapshotAssertion,
    information_vehicles,
    vehicle_2,
    position,
    location,
    capabilities,
) -> None:
    """Test connection."""
    api = AudiConnect(
        session=ClientSession(), username=USR, password=PWD, country=COUNTRY, spin=SPIN
    )

    with (
        patch(
            "audiconnectpy.api.AudiConnect.async_get_information_vehicles",
            return_value=information_vehicles,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_selectivestatus",
            return_value=vehicle_2,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_position",
            return_value=position,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_location",
            return_value=location,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_capabilities",
            return_value=capabilities,
        ),
    ):
        await api.async_login()
        my_vehicle = api.vehicles[0]

        assert api.vehicles is not None
        assert my_vehicle.infos.to_dict() == snapshot
        assert my_vehicle.user_capabilities.to_dict() == snapshot
        assert my_vehicle.access.to_dict() == snapshot
        assert my_vehicle.charging.to_dict() == snapshot
        assert my_vehicle.climatisation_timers.to_dict() == snapshot
        assert my_vehicle.climatisation.to_dict() == snapshot
        assert my_vehicle.fuel_status.to_dict() == snapshot
        assert my_vehicle.vehicle_health_inspection.to_dict() == snapshot
        assert my_vehicle.vehicle_lights.to_dict() == snapshot
        assert my_vehicle.measurements.to_dict() == snapshot
        # assert my_vehicle.oil_level.to_dict() == snapshot
        # assert my_vehicle.vehicle_health_warnings.to_dict() == snapshot


@pytest.mark.asyncio
@patch("audiconnectpy.auth.Auth.async_connect")
@patch("audiconnectpy.api.AudiConnect._async_fill_url")
async def test_vehicle_3(
    connect,
    fill_url,
    snapshot: SnapshotAssertion,
    information_vehicles,
    vehicle_3,
    position,
    location,
    capabilities,
) -> None:
    """Test connection."""
    api = AudiConnect(
        session=ClientSession(), username=USR, password=PWD, country=COUNTRY, spin=SPIN
    )

    with (
        patch(
            "audiconnectpy.api.AudiConnect.async_get_information_vehicles",
            return_value=information_vehicles,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_selectivestatus",
            return_value=vehicle_3,
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_position",
            side_effect=AudiException(),
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_location",
            side_effect=AudiException(),
        ),
        patch(
            "audiconnectpy.vehicle.Vehicle.async_get_capabilities",
            return_value=capabilities,
        ),
    ):
        await api.async_login()
        my_vehicle = api.vehicles[0]

        assert api.vehicles is not None
        assert my_vehicle.climatisation_timers.to_dict() == snapshot
        assert my_vehicle.climatisation.to_dict() == snapshot
        assert my_vehicle.charging.to_dict() == snapshot
        assert my_vehicle.position_supported is False
        assert my_vehicle.locations_supported is False
        assert my_vehicle.capabilities_supported is True
