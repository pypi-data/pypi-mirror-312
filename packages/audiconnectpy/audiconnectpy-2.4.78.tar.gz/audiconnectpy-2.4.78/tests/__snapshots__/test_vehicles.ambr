# serializer version: 1
# name: test_vehicle_0
  dict({
    'core': dict({
      'model_year': 2022,
    }),
    'media': dict({
      'long_name': 'Audi A4 Berline S line S Edition 35 TFSI 110(150) kW(ch) S tronic',
      'short_name': 'Audi A4 Berline',
    }),
  })
# ---
# name: test_vehicle_0.1
  dict({
    'access_status': dict({
      'car_captured_timestamp': '2024-05-11T16:41:40+00:00',
      'door_lock_status': False,
      'doors': Doors(locked=DoorLocked(left_front=True, right_front=True, left_rear=True, right_rear=True, trunk=True, doors_trunk=True, any_doors_status=True), opened=DoorOpened(left_front=False, right_front=False, left_rear=False, right_rear=False, trunk=False, bonnet=False, any_doors_status=False)),
      'overall_status': 'unsafe',
      'windows': Window(left_front=False, right_front=False, left_rear=False, right_rear=False, roof_cover=None, sun_roof=None, any_windows_status=False),
    }),
  })
# ---
# name: test_vehicle_0.2
  dict({
    'range_status': dict({
      'car_type': 'gasoline',
      'primary_engine': dict({
        'current_fuel_level_pct': 100,
        'current_soc_pct': 100,
        'remaining_range_km': 860,
        'type': 'gasoline',
      }),
      'secondary_engine': None,
      'total_range_km': 860,
    }),
  })
# ---
# name: test_vehicle_0.3
  dict({
    'maintenance_status': dict({
      'inspection_due_days': 131,
      'inspection_due_km': 8500,
      'mileage_km': 21512,
      'oil_service_due_days': 131,
      'oil_service_due_km': 8500,
    }),
  })
# ---
# name: test_vehicle_0.4
  dict({
    'lights_status': dict({
      'lights': Lights(left=False, right=False, status=False),
    }),
  })
# ---
# name: test_vehicle_0.5
  dict({
    'fuel_level_status': dict({
      'car_type': 'gasoline',
      'current_fuel_level_pct': 100,
      'current_soc_pct': None,
      'primary_engine_type': 'gasoline',
      'secondary_engine_type': None,
    }),
    'odometer_status': dict({
      'odometer': 21512,
    }),
    'range_status': dict({
      'electric_range': None,
      'gasoline_range': 860,
      'total_range_km': 860,
    }),
    'temperature_battery_status': None,
  })
# ---
# name: test_vehicle_0.6
  dict({
    'oil_level_status': dict({
      'value': True,
    }),
  })
# ---
# name: test_vehicle_0.7
  dict({
    'warning_lights': dict({
      'lights': None,
    }),
  })
# ---
# name: test_vehicle_1
  dict({
    'core': dict({
      'model_year': 2022,
    }),
    'media': dict({
      'long_name': 'Audi A4 Berline S line S Edition 35 TFSI 110(150) kW(ch) S tronic',
      'short_name': 'Audi A4 Berline',
    }),
  })
# ---
# name: test_vehicle_1.1
  dict({
    'capabilities_status': list([
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'webApp',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'automation',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'batteryChargingCare',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'batteryColdWarning',
        'userDisablingAllowed': True,
      }),
      dict({
        'id': 'batterySupport',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'charging',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'chargingProfiles',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'chargingStations',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'climatisation',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'climatisationTimers',
        'userDisablingAllowed': False,
      }),
      dict({
        'id': 'cubicNetwork',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'dealerAppointment',
        'status': list([
          1007,
          1002,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'destinations',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'fuelStatus',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'ignition',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'mapUpdate',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'measurements',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'onlineSpeech',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'parkingBrake',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'parkingInformation',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'parkingPosition',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'personalizationOnline',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'plugAndCharge',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'poiSearch',
        'userDisablingAllowed': True,
      }),
      dict({
        'id': 'readiness',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'routing',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'state',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'trafficInformation',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'vehicleHealth',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'vehicleHealthInspection',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'vehicleHealthWarnings',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'vehicleLights',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2082-12-25T23:59:59Z',
        'id': 'vehicleWakeUpTrigger',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2028-12-25T23:59:59Z',
        'id': 'webRadio',
        'userDisablingAllowed': True,
      }),
    ]),
  })
# ---
# name: test_vehicle_1.2
  dict({
    'access_status': dict({
      'car_captured_timestamp': '2024-03-25T10:29:31.596000+00:00',
      'door_lock_status': True,
      'doors': Doors(locked=DoorLocked(left_front=False, right_front=False, left_rear=False, right_rear=False, trunk=False, doors_trunk=False, any_doors_status=False), opened=DoorOpened(left_front=False, right_front=False, left_rear=False, right_rear=False, trunk=False, bonnet=False, any_doors_status=False)),
      'overall_status': 'safe',
      'windows': Window(left_front=False, right_front=False, left_rear=False, right_rear=False, roof_cover=None, sun_roof=None, any_windows_status=False),
    }),
  })
# ---
# name: test_vehicle_1.3
  dict({
    'battery_status': dict({
      'cruising_range_electric_km': 147,
      'current_soc_pct': 37,
    }),
    'charge_mode': dict({
      'available_charge_modes': list([
        'manual',
      ]),
      'preferred_charge_mode': 'manual',
    }),
    'charging_settings': dict({
      'auto_unlock_plug_when_charged': False,
      'auto_unlock_plug_when_charged_ac': False,
      'max_charge_current_ac': 'maximum',
      'target_soc_pct': 80,
    }),
    'charging_status': dict({
      'charge_mode': 'manual',
      'charge_power_kw': None,
      'charge_rate_kmph': 0,
      'charge_type': 'ac',
      'charging_settings': 'default',
      'charging_state': True,
      'remaining': 0,
    }),
    'plug_status': dict({
      'external_power': True,
      'led_color': 'green',
      'plug_connection_state': True,
      'plug_lock_state': True,
    }),
  })
# ---
# name: test_vehicle_1.4
  dict({
    'climatisation_timers_status': dict({
      'time_in_car': '2024-03-25T11:29:32+01:00',
      'timers': list([
        dict({
          'enabled': False,
          'id': 1,
          'single_timer': dict({
            'start': '2024-01-28T07:30:00+00:00',
            'start_local': None,
            'target': '2024-01-28T07:30:00+00:00',
            'target_local': None,
          }),
        }),
        dict({
          'enabled': False,
          'id': 2,
          'single_timer': dict({
            'start': '2023-10-12T06:20:00+00:00',
            'start_local': None,
            'target': '2023-10-12T06:20:00+00:00',
            'target_local': None,
          }),
        }),
      ]),
    }),
  })
# ---
# name: test_vehicle_1.5
  dict({
    'climatisation_settings': dict({
      'climatization_at_unlock': False,
      'target_temperature_c': 21,
      'target_temperature_f': 70,
      'unit_in_car': 'celsius',
      'window_heating_enabled': False,
      'zone_front_left_enabled': False,
      'zone_front_right_enabled': False,
      'zone_rear_left_enabled': None,
      'zone_rear_right_enabled': None,
    }),
    'climatisation_status': dict({
      'climatisation_state': 'off',
      'remaining_climatisation_time_min': 0,
    }),
    'window_heating_status': dict({
      'state': WindowHeating(front=False, rear=False),
    }),
  })
# ---
# name: test_vehicle_1.6
  dict({
    'range_status': dict({
      'car_type': 'electric',
      'primary_engine': dict({
        'current_fuel_level_pct': None,
        'current_soc_pct': 37,
        'remaining_range_km': 147,
        'type': 'electric',
      }),
      'secondary_engine': None,
      'total_range_km': 147,
    }),
  })
# ---
# name: test_vehicle_1.7
  dict({
    'maintenance_status': dict({
      'inspection_due_days': 261,
      'inspection_due_km': None,
      'mileage_km': 16221,
      'oil_service_due_days': None,
      'oil_service_due_km': None,
    }),
  })
# ---
# name: test_vehicle_1.8
  dict({
    'lights_status': dict({
      'lights': Lights(left=False, right=False, status=False),
    }),
  })
# ---
# name: test_vehicle_1.9
  dict({
    'fuel_level_status': dict({
      'car_type': 'electric',
      'current_fuel_level_pct': None,
      'current_soc_pct': None,
      'primary_engine_type': 'electric',
      'secondary_engine_type': None,
    }),
    'odometer_status': dict({
      'odometer': 16221,
    }),
    'range_status': dict({
      'electric_range': None,
      'gasoline_range': None,
      'total_range_km': None,
    }),
    'temperature_battery_status': dict({
      'temperature_hv_battery_max_k': 283.15,
      'temperature_hv_battery_min_k': 281.65,
    }),
  })
# ---
# name: test_vehicle_2
  dict({
    'core': dict({
      'model_year': 2022,
    }),
    'media': dict({
      'long_name': 'Audi A4 Berline S line S Edition 35 TFSI 110(150) kW(ch) S tronic',
      'short_name': 'Audi A4 Berline',
    }),
  })
# ---
# name: test_vehicle_2.1
  dict({
    'capabilities_status': list([
      dict({
        'id': 'webApp',
        'status': list([
          1013,
          1014,
          2003,
        ]),
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'automation',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'batteryChargingCare',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'batteryColdWarning',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'id': 'batterySupport',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'charging',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'chargingProfiles',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'chargingStations',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'climatisation',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'climatisationTimers',
        'userDisablingAllowed': False,
      }),
      dict({
        'id': 'cubicNetwork',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'dealerAppointment',
        'status': list([
          1003,
          1004,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'destinations',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'fuelStatus',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'ignition',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'mapUpdate',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'measurements',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'onlineSpeech',
        'status': list([
          3004,
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'parkingBrake',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'parkingInformation',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'parkingPosition',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'personalizationOnline',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'plugAndCharge',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'poiSearch',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2033-04-19T23:59:59Z',
        'id': 'privateEmergencyCall',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'id': 'readiness',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'roadsideAssistant',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'routing',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'state',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'trafficInformation',
        'status': list([
          1013,
          1014,
        ]),
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'vehicleHealth',
        'userDisablingAllowed': True,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'vehicleHealthInspection',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'vehicleHealthWarnings',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'vehicleLights',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'vehicleWakeUpTrigger',
        'userDisablingAllowed': False,
      }),
      dict({
        'expirationDate': '2025-04-19T23:59:59Z',
        'id': 'webRadio',
        'userDisablingAllowed': True,
      }),
    ]),
  })
# ---
# name: test_vehicle_2.2
  dict({
    'access_status': dict({
      'car_captured_timestamp': '2023-12-23T19:36:52.444000+00:00',
      'door_lock_status': True,
      'doors': Doors(locked=DoorLocked(left_front=False, right_front=False, left_rear=False, right_rear=False, trunk=False, doors_trunk=False, any_doors_status=False), opened=DoorOpened(left_front=False, right_front=False, left_rear=False, right_rear=False, trunk=False, bonnet=False, any_doors_status=False)),
      'overall_status': 'safe',
      'windows': Window(left_front=False, right_front=False, left_rear=False, right_rear=False, roof_cover=None, sun_roof=None, any_windows_status=False),
    }),
  })
# ---
# name: test_vehicle_2.3
  dict({
    'battery_status': dict({
      'cruising_range_electric_km': 212,
      'current_soc_pct': 56,
    }),
    'charge_mode': dict({
      'available_charge_modes': list([
        'manual',
        'preferredChargingTimes',
      ]),
      'preferred_charge_mode': 'manual',
    }),
    'charging_settings': dict({
      'auto_unlock_plug_when_charged': False,
      'auto_unlock_plug_when_charged_ac': False,
      'max_charge_current_ac': 'maximum',
      'target_soc_pct': 100,
    }),
    'charging_status': dict({
      'charge_mode': 'manual',
      'charge_power_kw': None,
      'charge_rate_kmph': 27,
      'charge_type': 'ac',
      'charging_settings': 'profile',
      'charging_state': True,
      'remaining': 400,
    }),
    'plug_status': dict({
      'external_power': True,
      'led_color': 'green',
      'plug_connection_state': True,
      'plug_lock_state': True,
    }),
  })
# ---
# name: test_vehicle_2.4
  dict({
    'climatisation_timers_status': dict({
      'time_in_car': '2023-12-23T20:36:53+01:00',
      'timers': list([
        dict({
          'enabled': False,
          'id': 1,
          'single_timer': dict({
            'start': '2023-11-28T06:40:00+00:00',
            'start_local': None,
            'target': '2023-11-28T06:40:00+00:00',
            'target_local': None,
          }),
        }),
        dict({
          'enabled': False,
          'id': 2,
          'single_timer': dict({
            'start': '1999-12-31T23:00:00+00:00',
            'start_local': None,
            'target': '1999-12-31T23:00:00+00:00',
            'target_local': None,
          }),
        }),
      ]),
    }),
  })
# ---
# name: test_vehicle_2.5
  dict({
    'climatisation_settings': dict({
      'climatization_at_unlock': True,
      'target_temperature_c': 22,
      'target_temperature_f': 72,
      'unit_in_car': 'celsius',
      'window_heating_enabled': True,
      'zone_front_left_enabled': False,
      'zone_front_right_enabled': False,
      'zone_rear_left_enabled': None,
      'zone_rear_right_enabled': None,
    }),
    'climatisation_status': dict({
      'climatisation_state': 'off',
      'remaining_climatisation_time_min': 0,
    }),
    'window_heating_status': dict({
      'state': WindowHeating(front=False, rear=False),
    }),
  })
# ---
# name: test_vehicle_2.6
  dict({
    'range_status': dict({
      'car_type': 'electric',
      'primary_engine': dict({
        'current_fuel_level_pct': None,
        'current_soc_pct': 56,
        'remaining_range_km': 212,
        'type': 'electric',
      }),
      'secondary_engine': None,
      'total_range_km': 212,
    }),
  })
# ---
# name: test_vehicle_2.7
  dict({
    'maintenance_status': dict({
      'inspection_due_days': 478,
      'inspection_due_km': None,
      'mileage_km': 10329,
      'oil_service_due_days': None,
      'oil_service_due_km': None,
    }),
  })
# ---
# name: test_vehicle_2.8
  dict({
    'lights_status': dict({
      'lights': Lights(left=False, right=False, status=False),
    }),
  })
# ---
# name: test_vehicle_2.9
  dict({
    'fuel_level_status': dict({
      'car_type': 'electric',
      'current_fuel_level_pct': None,
      'current_soc_pct': 56,
      'primary_engine_type': 'electric',
      'secondary_engine_type': None,
    }),
    'odometer_status': dict({
      'odometer': 10329,
    }),
    'range_status': dict({
      'electric_range': 212,
      'gasoline_range': None,
      'total_range_km': 212,
    }),
    'temperature_battery_status': None,
  })
# ---
# name: test_vehicle_3
  dict({
    'climatisation_timers_status': dict({
      'time_in_car': '2023-12-23T20:36:53+01:00',
      'timers': list([
        dict({
          'enabled': False,
          'id': 1,
          'single_timer': dict({
            'start': None,
            'start_local': '2023-11-28T06:40:00+00:00',
            'target': None,
            'target_local': '2023-11-28T06:40:00+00:00',
          }),
        }),
        dict({
          'enabled': False,
          'id': 2,
          'single_timer': dict({
            'start': None,
            'start_local': '1999-12-31T23:00:00+00:00',
            'target': None,
            'target_local': '1999-12-31T23:00:00+00:00',
          }),
        }),
      ]),
    }),
  })
# ---
# name: test_vehicle_3.1
  dict({
    'climatisation_settings': dict({
      'climatization_at_unlock': True,
      'target_temperature_c': 22,
      'target_temperature_f': 72,
      'unit_in_car': 'celsius',
      'window_heating_enabled': True,
      'zone_front_left_enabled': False,
      'zone_front_right_enabled': False,
      'zone_rear_left_enabled': None,
      'zone_rear_right_enabled': None,
    }),
    'climatisation_status': dict({
      'climatisation_state': 'heating',
      'remaining_climatisation_time_min': 0,
    }),
    'window_heating_status': dict({
      'state': WindowHeating(front=False, rear=False),
    }),
  })
# ---
# name: test_vehicle_3.2
  dict({
    'battery_status': dict({
      'cruising_range_electric_km': 212,
      'current_soc_pct': 56,
    }),
    'charge_mode': dict({
      'available_charge_modes': list([
        'manual',
        'preferredChargingTimes',
      ]),
      'preferred_charge_mode': 'manual',
    }),
    'charging_settings': dict({
      'auto_unlock_plug_when_charged': False,
      'auto_unlock_plug_when_charged_ac': False,
      'max_charge_current_ac': 'maximum',
      'target_soc_pct': 100,
    }),
    'charging_status': dict({
      'charge_mode': 'manual',
      'charge_power_kw': None,
      'charge_rate_kmph': 27,
      'charge_type': 'ac',
      'charging_settings': 'profile',
      'charging_state': True,
      'remaining': 400,
    }),
    'plug_status': dict({
      'external_power': False,
      'led_color': 'none',
      'plug_connection_state': False,
      'plug_lock_state': False,
    }),
  })
# ---
